consensus/tests/checkpoint.rs

```diff
@@ -43,48 +47,194 @@ mod tests {
-        b2.body.consensus_state.curr_global_slot = GlobalSlot {
-            slot_number: GlobalSlotNumber(667),
-            slots_per_epoch: Length(7140),
-        };
-        c3.push(b2).unwrap();
-        assert_eq!(is_short_range(&c3, &c0).unwrap(), false);
-        assert_eq!(is_short_range(&c0, &c3).unwrap(), false);
+
+        let mut a: ProtocolState = Default::default();
+        let mut b: ProtocolState = Default::default();
+
+        a.body.consensus_state = ConsensusState::default();
+        b.body.consensus_state = ConsensusState::default();
+        gen_spot_pair_common_checkpoints(&mut a, &mut b, 0);
+
+        let c0: ProtocolStateChain = ProtocolStateChain(vec![a]);
+        let c1: ProtocolStateChain = ProtocolStateChain(vec![b]);
+
+        assert_eq!(is_short_range(&c0, &c1).unwrap(), true);
+    }
+
+    #[test]
+    #[wasm_bindgen_test]
+    fn gen_spot_pair_short_misaligned_generates_pairs_of_states_in_short_fork_range() {
+        let mut genesis: ProtocolState = Default::default();
+        init_checkpoints(&mut genesis).unwrap();
+
+        let mut a: ProtocolState = Default::default();
+        let mut b: ProtocolState = Default::default();
+
+        a.body.consensus_state = ConsensusState::default();
+        b.body.consensus_state = ConsensusState::default();
+
+        // Compute the root epoch position of `b`. This needs to be one epoch ahead of a
+        let added_blocks =
+            gen_num_blocks_in_slots(DEFAULT_SLOT_FILL_RATE, DEFAULT_SLOT_FILL_RATE_DELTA, 1.0);
+
+        b.body.consensus_state.blockchain_length.0 =
+            a.body.consensus_state.blockchain_length.0 + added_blocks as u32;
+
+        // Constrain first state to be within last 1/3rd of its epoch (ensuring it's checkpoints and seed are fixed)
+        let protocol_constants = ConsensusConstants::default();
+        let min_a_curr_epoch_slot = 2 * (protocol_constants.slots_per_epoch.0 / 3) + 1;
+        gen_spot_pair_common_checkpoints(&mut a, &mut b, min_a_curr_epoch_slot);
+
+        let c0: ProtocolStateChain = ProtocolStateChain(vec![a]);
+        let c1: ProtocolStateChain = ProtocolStateChain(vec![b]);
+
+        assert_eq!(is_short_range(&c0, &c1).unwrap(), true);
+    }
+
+    #[test]
+    #[wasm_bindgen_test]
+    fn gen_spot_pair_long_generates_pairs_of_states_in_long_fork_range() {
+        let mut genesis: ProtocolState = Default::default();
+        init_checkpoints(&mut genesis).unwrap();
+
+        let mut a: ProtocolState = Default::default();
+        let mut b: ProtocolState = Default::default();
+
+        a.body.consensus_state = ConsensusState::default();
+        b.body.consensus_state = ConsensusState::default();
+
+        a.body.consensus_state.epoch_count = Length(14);
+        b.body.consensus_state.epoch_count = Length(15);
+        let c0: ProtocolStateChain = ProtocolStateChain(vec![a]);
+        let c1: ProtocolStateChain = ProtocolStateChain(vec![b]);
+
```

Comment:
shouldn't the slot numbers (`epoch_slot`) be set here as well? won't they have the same slot number here?

Reply:
No, we already modify into this [line ](https://github.com/ChainSafe/mina-rs/blob/38edaa280d2c25f714481b908b9fb01209b67d2b/base/src/consensus_state.rs#L191) not use the previous way.